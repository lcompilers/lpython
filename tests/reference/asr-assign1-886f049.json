{
    "basename": "asr-assign1-886f049",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/assign1.py",
    "infile_hash": "3b82a73e457bd65e85828b72d56596ca927e7c661e333691f154912b",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-assign1-886f049.stdout",
    "stdout_hash": "c2ffd64d63b0780c568d6f28ec8306cade4e74588adc6f7db056603f",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}