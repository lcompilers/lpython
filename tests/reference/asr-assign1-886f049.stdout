(TranslationUnit
    (SymbolTable
        1
        {
            __main__:
                (Module
                    (SymbolTable
                        2
                        {
                            test_augassign:
                                (Function
                                    (SymbolTable
                                        3
                                        {
                                            a:
                                                (Variable
                                                    3
                                                    a
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Character 1 -2 ())
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            r:
                                                (Variable
                                                    3
                                                    r
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            s:
                                                (Variable
                                                    3
                                                    s
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    test_augassign
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(Assignment
                                        (Var 3 r)
                                        (IntegerConstant 0 (Integer 4))
                                        ()
                                    )
                                    (Assignment
                                        (Var 3 r)
                                        (IntegerBinOp
                                            (Var 3 r)
                                            Add
                                            (IntegerConstant 4 (Integer 4))
                                            (Integer 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assignment
                                        (Var 3 s)
                                        (IntegerConstant 5 (Integer 4))
                                        ()
                                    )
                                    (Assignment
                                        (Var 3 r)
                                        (IntegerBinOp
                                            (Var 3 r)
                                            Mul
                                            (Var 3 s)
                                            (Integer 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assignment
                                        (Var 3 r)
                                        (IntegerBinOp
                                            (Var 3 r)
                                            Sub
                                            (IntegerConstant 2 (Integer 4))
                                            (Integer 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assignment
                                        (Var 3 s)
                                        (IntegerConstant 10 (Integer 4))
                                        ()
                                    )
                                    (Assignment
                                        (Var 3 r)
                                        (RealBinOp
                                            (Cast
                                                (Var 3 r)
                                                IntegerToReal
                                                (Real 8)
                                                ()
                                            )
                                            Div
                                            (Cast
                                                (Var 3 s)
                                                IntegerToReal
                                                (Real 8)
                                                ()
                                            )
                                            (Real 8)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assignment
                                        (Var 3 a)
                                        (StringConstant
                                            ""
                                            (Character 1 0 ())
                                        )
                                        ()
                                    )
                                    (Assignment
                                        (Var 3 a)
                                        (StringConcat
                                            (Var 3 a)
                                            (StringConstant
                                                "test"
                                                (Character 1 4 ())
                                            )
                                            (Character 1 2 ())
                                            ()
                                        )
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    __main__
                    []
                    .false.
                    .false.
                ),
            main_program:
                (Program
                    (SymbolTable
                        4
                        {
                            
                        })
                    main_program
                    []
                    []
                )
        })
    []
)
